### PR TITLE
[Test] Query dashboard to get the serve application status in head pod

### LINKED ISSUE #2481: Query dashboard to get the serve application status in head pod
Ref:
- https://github.com/ray-project/kuberay/blob/5cb2f56b9d3eebbadbeb0aec70dfd30a9d5dab83/ray-operator/controllers/ray/rayservice_controller.go#L870

### PR DESCRIPTION
**Pull request by CheyuWu**
## Why are these changes needed?
- Add the testcase to query dashboard to get the serve application status in head pod

## Related issue number
Closes #2481 

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [X] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by CheyuWu** [comment]
@MortalHappiness PTAL

**Review by MortalHappiness — CHANGES_REQUESTED** [review]
Your implementation is too complicated. Please reference this:

https://github.com/ray-project/kuberay/blob/7e04b2201b73f9959bc4fc775cce444f62094b98/ray-operator/controllers/ray/utils/dashboard_httpclient.go#L162-L196

You only need to send a GET request to dashboard URL and then check it's response.

File: ray-operator/test/sampleyaml/support.go [hunk 1]
```diff
@@ -101,3 +103,24 @@ func SubmitJobsToAllPods(t Test, rayCluster *rayv1.RayCluster) func(Gomega) {
...
+		remotePort := 8265
+		stopChan, err := SetupPortForward(t, pod.Name, pod.Namespace, localPort, remotePort)
+		defer close(stopChan)
+
+		g.Expect(err).ToNot(HaveOccurred())
+		url := fmt.Sprintf("127.0.0.1:%d", localPort)
+
+		err = rayDashboardClient.InitClient(t.Ctx(), url, rayCluster)
+		g.Expect(err).ToNot(HaveOccurred())
+		_, err = rayDashboardClient.GetServeDetails(t.Ctx())
```
Comment (kevin85421) [inline ray-operator/test/sampleyaml/support.go:123]: Can you also check whether all Serve application statuses on the Ray dashboard are `rayv1.ApplicationStatusEnum.RUNNING`?

**Comment by kevin85421** [comment]
btw, can you rebase with the master branch?

File: ray-operator/test/sampleyaml/support.go (hunk 1 above)
Comment (MortalHappiness) [reply ray-operator/test/sampleyaml/support.go:123]: @kevin85421 I think this has been addressed in https://github.com/ray-project/kuberay/pull/2496 right?

Comment (kevin85421) [reply ray-operator/test/sampleyaml/support.go:123]: They are different. #2496 checks RayService status, and this PR checks Ray dashboard which is the source of the truth.

Comment (CheyuWu) [reply ray-operator/test/sampleyaml/support.go]: OK, No problem I will handle that.

**Comment by MortalHappiness** [comment]
@CheyuWu You don't need to fix it because it is very flaky and will be removed after all sample yaml refactoring subtasks are done.

**Review by kevin85421 — APPROVED** [review]

**Review by MortalHappiness — APPROVED** [review]

**Pull request merged**